src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesNoHttpTool.java

```diff
@@ -0,0 +1,159 @@
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHFileNotFoundException;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * This data provider checks if an open-source project uses
+ * <a href="https://github.com/spring-io/nohttp">nohttp</a> tool to make sure that
+ * plain HTTP is not used.
+ * The provider fills out the
+ * {@link com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures#USES_NOHTTP} feature.
+ */
+public class UsesNoHttpTool extends AbstractGitHubDataProvider {
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param github An interface to the GitHub API.
+   */
+  public UsesNoHttpTool(GitHub github) {
+    super(github);
+  }
+
+  @Override
+  protected UsesNoHttpTool doUpdate(GitHubProject project, ValueSet values) throws IOException {
+    logger.info("Figuring out if the project uses NoHTTP ...");
+
+    GHRepository repository = gitHubDataFetcher().repositoryFor(project, github);
+    boolean answer = checkMaven(repository) || checkGradle(repository);
+    Value<Boolean> value = new BooleanValue(USES_NOHTTP, answer);
+    values.update(value);
+
+    return this;
+  }
+
+  /**
+   * Checks if a project uses NoHTTP with Maven.
+   *
+   * @param repository The project's repository.
+   * @return True if the project uses the plugin, false otherwise.
+   */
+  private boolean checkMaven(GHRepository repository) throws IOException {
+    GHContent content;
+    try {
+      content = repository.getFileContent("pom.xml");
+    } catch (GHFileNotFoundException e) {
+      return false;
+    }
+
+    if (content == null || !content.isFile()) {
+      return false;
+    }
+
+    Model model = readModel(content);
+
+    if (model.getBuild() != null) {
+      for (Plugin plugin : model.getBuild().getPlugins()) {
+        if (isNoHttp(plugin)) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if a project uses nohttp with Gradle.
+   *
+   * @param repository The project's repository.
+   * @return True if the project uses the plugin, false otherwise.
+   */
+  private boolean checkGradle(GHRepository repository) throws IOException {
+    GHContent content;
+    try {
+      content = repository.getFileContent("build.gradle");
+    } catch (GHFileNotFoundException e) {
+      return false;
+    }
+
+    if (content == null || !content.isFile()) {
+      return false;
+    }
+
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(content.read()))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        if (line.trim().startsWith("id \"io.spring.nohttp\"")) {
```

Comment:
Can the string be a final static?

Reply:
Yes, it can. Since it's used only in one place, let's keep it as-is. I think it makes it a bit easier to read code since you don't have to jump to the definition of the `final static`.